editor/editor_command_palette.cpp

```diff
@@ -0,0 +1,246 @@
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "editor/editor_command_palette.h"
+#include "core/os/keyboard.h"
+#include "editor/editor_node.h"
+#include "scene/gui/tree.h"
+
+EditorCommandPalette *EditorCommandPalette::singleton = nullptr;
+
+float EditorCommandPalette::_score_path(const String &p_search, const String &p_path) {
+	float score = 0.9f + .1f * (p_search.length() / (float)p_path.length());
+
+	// Positive bias for matches close to the beginning of the file name.
+	int pos = p_path.findn(p_search);
+	if (pos != -1) {
+		return score * (1.0f - 0.1f * (float(pos) / p_path.length()));
+	}
+
+	// Positive bias for matches close to the end of the path.
+	pos = p_path.rfindn(p_search);
+	if (pos != -1) {
+		return score * (0.8f - 0.1f * (float(p_path.length() - pos) / p_path.length()));
+	}
+
+	// Remaining results belong to the same class of results.
+	return score * 0.69f;
+}
+
+void EditorCommandPalette::_update_command_search() {
+	commands.get_key_list(command_keys);
+
+	if (command_keys != nullptr) {
+		const String search_text = command_search_box->get_text();
+		const bool empty_search = search_text.is_empty();
+
+		// Filter possible candidates.
+		Vector<CommandEntry> entries;
+		for (int i = 0; i < command_keys->size(); i++) {
+			CommandEntry r;
+			r.key_name = (*command_keys)[i];
+			r.display_name = commands[r.key_name].name;
+			if (!empty_search && search_text.is_subsequence_ofi(r.display_name)) {
+				r.score = empty_search ? 0 : _score_path(search_text, r.display_name.to_lower());
+				entries.push_back(r);
+			}
+		}
+
+		command_keys->clear();
+
+		TreeItem *root = search_options->get_root();
+		root->clear_children();
+
+		if (entries.size() > 0) {
+			if (!empty_search) {
+				SortArray<CommandEntry, CommandEntryComparator> sorter;
+				sorter.sort(entries.ptrw(), entries.size());
+			}
+
+			const int entry_limit = MIN(entries.size(), 300);
+			for (int i = 0; i < entry_limit; i++) {
+				TreeItem *ti = search_options->create_item(root);
+				ti->set_tooltip(0, entries[i].key_name);
+				ti->set_text(0, entries[i].display_name);
+			}
+
+			TreeItem *to_select = root->get_first_child();
+			to_select->select(0);
+			to_select->set_as_cursor(0);
+			search_options->scroll_to_item(to_select);
+
+			get_ok_button()->set_disabled(false);
+		} else {
+			search_options->deselect_all();
+
+			get_ok_button()->set_disabled(true);
+		}
+	} else {
+		ERR_FAIL_COND(!command_keys);
```

Comment:
No, you should remove the `if (command_keys != nullptr) {` and move `ERR_FAIL_COND(!command_keys);` at the top of your code.

Reply:
oops... 😅